File: string_processing/py/string_joins_01.py
#!/usr/bin/python

####################################
# Py joins on sequences
# author: vladimir kulyukin
####################################

seq1 = ['your', 'thinking', 'is', 'like', 'a', 'camel', 'driver']
seq2 = ['a', 'b', 'c', 'd', 'e', 'f']

def join_seq(separator, seq):
  return separator.join(seq)

def test_join_seq(seq):
  print join_seq('*',    seq)
  print join_seq(' ** ', seq)
  print join_seq('//',   seq)

test_join_seq(seq1)
test_join_seq(seq2)

File: string_processing/pl/string_joins_01.pl
#!/usr/bin/perl

###############################################
# joins on sequences in PL
# author: vladimir kulyukin
###############################################

use strict;
use warnings;

my @seq1 = qw(your thinking is like a camel driver);
my @seq2 = ('a' .. 'f');

sub join_seq {
    my ($separator, $seq) = @_;
    return join($separator, @{$seq});
}

sub test_join_seq {
    print join_seq('*',    \@_), "\n";
    print join_seq(' ** ', \@_), "\n";
    print join_seq('//',   \@_), "\n";
}

test_join_seq(@seq1);
test_join_seq(@seq2);
